src/global/signal_handler.cc

```diff
@@ -246,10 +249,20 @@ struct SignalHandler : public Thread {
 	lock.Lock();
 	for (unsigned signum=0; signum<32; signum++) {
 	  if (handlers[signum]) {
-	    r = read(handlers[signum]->pipefd[0], &v, 1);
-	    if (r == 1) {
-	      handlers[signum]->handler(signum);
-	    }
+	   if(true==handlers[signum]->is_siginfo)
+        {
+            r = read(handlers[signum]->pipefd[0], &v, 1);
```

Comment:
You may want to put this line out of this 'if' block as it's the same for both true/false cases